### PR TITLE
profiler: Use BPF batch operations

### PR DESCRIPTION
**Pull request by kakkoyun**

### CODE REVIEW

**Comment by kakkoyun** [comment]
Blocked by https://github.com/aquasecurity/libbpfgo/issues/147

File: pkg/profiler/profiler.go [hunk 1]
```diff
@@ -334,77 +294,114 @@ func (p *CgroupProfiler) Run(ctx context.Context) error {
...
+	if err := it.Err(); err != nil {
+		return fmt.Errorf("iterate over counts map: %w", err)
+	}
 
-		r := bytes.NewBuffer(keyBytes)
+	if batchSize == 0 {
+		return nil
+	}
+	level.Debug(p.logger).Log("msg", "fetching stack trace counts in batch", "batchSize", batchSize)
+	time.Sleep(1 * time.Second)
```
Comment (kakkoyun) [inline pkg/profiler/profiler.go:356]: @brancz :scream:

Comment (v-thakkar) [reply pkg/profiler/profiler.go:356]: So profiler needs to wait until the BPF program gives the information about stack trace counts. Is that correct? In that case, do you think we can use something like SECCOMP_RET_USER_NOTIF (https://lwn.net/Articles/756233/)?

Comment (kakkoyun) [reply pkg/profiler/profiler.go:356]: This wait should've happened. We still don't know the reason why it works with it. I'm still investigating. Please see the linked issue. 

And thanks for the tip that might help us here, I'll check.

File: pkg/profiler/profiler.go [hunk 2]
```diff
@@ -334,77 +294,114 @@ func (p *CgroupProfiler) Run(ctx context.Context) error {
...
+			pid           = key.pid
+			userStackID   = key.userStackID
+			kernelStackID = key.kernelStackID
+		)
 
-		kernelStackIDBytes := make([]byte, 4)
-		if _, err := io.ReadFull(r, kernelStackIDBytes); err != nil {
-			return fmt.Errorf("read kernel stack ID bytes: %w", err)
+		if pid == 0 {
+			break
```
Comment (kakkoyun) [inline pkg/profiler/profiler.go:391]: @kakkoyun 
```suggestion
			continue
```

**Pull request closed without merging**